### PR TITLE
protonvpn-gui, protonvpn-cli: init at 1.0.1, 3.7.2

### PR DESCRIPTION
**Pull request by NKJe**
###### Motivation for this change
ProtonVPN has release two new interfaces for using their VPN service. This PR includes both, however, if requested it can be split. Furthermore it includes a suggestion for renaming the old packages protonvpn packages. This is suboptimal, but so are the alternatives. Solutions:
- The aforementioned solution: this results in users being forced to update their packages to this (possibly breaking), but allow them to used the old packages
- Make this PR a version bump to the old packages: this results in users being forced to update their packages to this (possibly breaking)
- Give these packages a new name possibly adding `official`, however this could still possibly confuse new users.

I will update the PR according to the wishes of the community. I've not currently added myself as maintainer, as the maintainers of the old packages might wish to continue to maintain this.

The PR also includes two direct dependencies, which are common for both packages. Hopefully this PR can act as a foundation for getting these packages in to nixpkgs.
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -26357,10 +26357,14 @@ in
 
   protonmail-bridge = callPackage ../applications/networking/protonmail-bridge { };
 
-  protonvpn-cli = callPackage ../applications/networking/protonvpn-cli { };
+  protonvpn-community-cli = callPackage ../applications/networking/protonvpn-community-cli { };
+
+  protonvpn-legacy-gui = callPackage ../applications/networking/protonvpn-legacy-gui { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:26362]: I think we can safely drop this.

File: pkgs/development/python-modules/protonvpn-nm-lib/default.nix [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    jinja2
+    distro
+    systemd
+  ];
+
+  meta = with lib; {
+    description = "ProtonVPN NM Library";
+    homepage = "https://github.com/ProtonVPN/protonvpn-nm-lib";
+    license = licenses.gpl3;
+    maintainers = [ ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/protonvpn-nm-lib/default.nix:64]: This must be set.

File: pkgs/development/python-modules/protonvpn-nm-lib/default.nix [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    pygobject3
+    jinja2
+    distro
+    systemd
+  ];
+
+  meta = with lib; {
+    description = "ProtonVPN NM Library";
+    homepage = "https://github.com/ProtonVPN/protonvpn-nm-lib";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/protonvpn-nm-lib/default.nix:63]: Do you know if the license is gpl3Only or gpl3Plus?

File: pkgs/development/python-modules/protonvpn-nm-lib/default.nix [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      substituteAll {
+        src = ./gi-path-patch.patch;
+        networkmanager_path = "${networkmanager}/lib/girepository-1.0";
+      }
+    )
+  ];
+
+  doCheck = false;
+
+  # pythonImportsCheck = [ "protonvpn_nm_lib.api" ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/protonvpn-nm-lib/default.nix:48]: Why is this disabled?

File: pkgs/development/python-modules/protonvpn-nm-lib/default.nix [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    )
+    (
+      substituteAll {
+        src = ./gi-path-patch.patch;
+        networkmanager_path = "${networkmanager}/lib/girepository-1.0";
+      }
+    )
+  ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/protonvpn-nm-lib/default.nix:46]: Please comment why tests are disabled.

File: pkgs/development/python-modules/protonvpn-nm-lib/default.nix [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        path = lib.makeBinPath [ networkmanager pkgs.systemd ncurses xdg-utils ];
+      }
+    )
+    (
+      substituteAll {
+        src = ./gi-path-patch.patch;
+        networkmanager_path = "${networkmanager}/lib/girepository-1.0";
+      }
+    )
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/protonvpn-nm-lib/default.nix:44]:
```suggestion
    (substituteAll {
      src = ./binary-paths.patch;
      path = lib.makeBinPath [ networkmanager pkgs.systemd ncurses xdg-utils ];
    })
    (substituteAll {
      src = ./gi-path-patch.patch;
      networkmanager_path = "${networkmanager}/lib/girepository-1.0";
    })
  ];
```

File: pkgs/development/python-modules/proton-client/default.nix [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+  propagatedBuildInputs = [ requests bcrypt python-gnupg pyopenssl ];
+
+  pythonImportsCheck = [ "proton.api" ];
+
+  meta = with lib; {
+    description = "Proton API Python Client";
+    homepage = "https://github.com/ProtonMail/proton-python-client";
+    license = licenses.gpl3;
+    maintainers = [ ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/proton-client/default.nix:44]: Do you know if the license is gpl3Only or gpl3Plus?
and maintainer needs to be set.

File: pkgs/development/python-modules/proton-client/default.nix [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  };
+
+  patches = [
+    (
+      substituteAll {
+        src = ./openssl-path.patch;
+        openssl = openssl.out;
+        ext = stdenv.hostPlatform.extensions.sharedLibrary;
+      }
+    )
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/proton-client/default.nix:33]:
```suggestion
    (substituteAll {
      src = ./openssl-path.patch;
      openssl = openssl.out;
      ext = stdenv.hostPlatform.extensions.sharedLibrary;
    })
```

File: pkgs/applications/networking/protonvpn-legacy-gui/default.nix [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+  '';
+
+  # no tests
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Linux GUI for ProtonVPN, written in Python";
+    homepage = "https://github.com/calexandru2018/linux-gui-legacy";
+    maintainers = with maintainers; [ offline ];
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/protonvpn-legacy-gui/default.nix:86]: Do you know if the license is gpl3Only or gpl3Plus?

File: pkgs/applications/networking/protonvpn-gui/default.nix [hunk 10]
```diff
@@ -1,89 +1,61 @@
...
-
   meta = with lib; {
-    description = "Linux GUI for ProtonVPN, written in Python";
-    homepage = "https://github.com/ProtonVPN/linux-gui";
-    maintainers = with maintainers; [ offline ];
+    description = "ProtonVPN Linux App";
+    homepage = "https://github.com/ProtonVPN/linux-app";
     license = licenses.gpl3;
-    platforms = platforms.linux;
+    maintainers = [ ];
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/protonvpn-gui/default.nix:59]: This needs to be set.

File: pkgs/applications/networking/protonvpn-gui/default.nix [hunk 11]
```diff
@@ -1,89 +1,61 @@
...
+{ lib
+, callPackage
+, python3Packages
+, fetchFromGitHub
+, wrapGAppsHook
+, gobject-introspection
+, imagemagick
+, gtk3
+, withIndicator ? true
+, libappindicator-gtk3 ? null
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/protonvpn-gui/default.nix:10]:
```suggestion
, libappindicator-gtk3
```

File: pkgs/applications/networking/protonvpn-community-cli/default.nix [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+    ];
+
+  # No tests
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Linux command-line client for ProtonVPN";
+    homepage = "https://github.com/ProtonVPN/linux-cli-community";
+    maintainers = with maintainers; [ jtcoolen jefflabonte shamilton ];
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/protonvpn-community-cli/default.nix:33]: Do you know if the license is gpl3Only or gpl3Plus?

File: pkgs/applications/networking/protonvpn-cli/default.nix [hunk 13]
```diff
@@ -1,36 +1,43 @@
...
 
   meta = with lib; {
-    description = "Linux command-line client for ProtonVPN";
-    homepage = "https://github.com/protonvpn/linux-cli";
-    maintainers = with maintainers; [ jtcoolen jefflabonte shamilton ];
+    description = "ProtonVPN Linux CLI";
+    homepage = "https://github.com/ProtonVPN/linux-cli";
     license = licenses.gpl3;
-    platforms = platforms.linux;
+    maintainers = [ ];
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/protonvpn-cli/default.nix:41]: This needs to be set.

File: pkgs/applications/networking/protonvpn-cli/default.nix [hunk 14]
```diff
@@ -1,36 +1,43 @@
...
-      pythondialog
-    ]) ++ [
-      dialog
-      openvpn
-      iptables
-    ];
-
-  # No tests
+  buildInputs = [ ];
+
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/protonvpn-cli/default.nix:20]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129333` at 7f727456 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built successfully:</summary>
  <ul>
    <li>protonvpn-cli</li>
    <li>protonvpn-community-cli</li>
    <li>protonvpn-gui</li>
    <li>protonvpn-legacy-gui</li>
    <li>python38Packages.proton-client</li>
    <li>python38Packages.protonvpn-nm-lib</li>
    <li>python39Packages.proton-client</li>
    <li>python39Packages.protonvpn-nm-lib</li>
  </ul>
</details>
<details>
  <summary>12 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/proton-client/default.nix:43:5:
```
   |
43 |     license = licenses.gpl3;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/python-modules/protonvpn-nm-lib/default.nix:32:5:
```
   |
32 |     (
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/proton-client/default.nix:44:5:
```
   |
44 |     maintainers = [ ];
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/protonvpn-nm-lib/default.nix:64:5:
```
   |
64 |     maintainers = [ ];
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/python-modules/proton-client/default.nix:27:5:
```
   |
27 |     (
   |     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `callPackage`.
Near pkgs/applications/networking/protonvpn-gui/default.nix:2:3:
```
  |
2 | , callPackage
  |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/python-modules/protonvpn-nm-lib/default.nix:38:5:
```
   |
38 |     (
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/protonvpn-nm-lib/default.nix:63:5:
```
   |
63 |     license = licenses.gpl3;
   |     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `pango`.
Near pkgs/applications/networking/protonvpn-legacy-gui/default.nix:2:66:
```
  |
2 | , python3Packages, wrapGAppsHook, protonvpn-community-cli, gtk3, pango
  |                                                                  ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/protonvpn-nm-lib/default.nix:61:0:
```
   |
61 |     description = "ProtonVPN NM Library";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/proton-client/default.nix:41:0:
```
   |
41 |     description = "Proton API Python Client";
   | ^
```
</li>
    <li>warning: unused-argument

Unused argument: `callPackage`.
Near pkgs/applications/networking/protonvpn-cli/default.nix:4:3:
```
  |
4 | , callPackage
  |   ^
```
</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (NKJe) [reply pkgs/top-level/all-packages.nix:26362]: Thanks for the review. To be sure what you suggest is to completely remove the "old packages"?

Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:26362]: I thought the new packages replace the old one. Is there a good reason to continue to use the old package if it is no longer being developed?

Comment (NKJe) [reply pkgs/top-level/all-packages.nix:26362]: I believe that they renamed it the repository to indicate that the old package would not be officially supported (I'm not sure to what extent is was officially supported previously). They've also recently included a notice in the old package, informing users on systems with networkmanager, about the new package. The only reason I'm slightly skeptical about removing the old packages is that I don't know the extent of the usage of these packages in the community. We could possibly give the maintainers of the old packages a some time to share their insights.

Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:26362]: Hmmm.. I am not sure what is the right thing todo here. People who do not use networkmanager can't use the new package?

**Comment by SuperSandro2000** [comment]
Can you please squash the review commits into the init comments of the packages?

**Comment by NKJe** [comment]
> Can you please squash the review commits into the init comments of the packages?

Done

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (NKJe) [reply pkgs/top-level/all-packages.nix:26362]: No they would be unable to. We could use the old names and add `official` at the end as such:
- `protonvpn-cli-official`
- `protonvpn-gui-official`

**Comment by NKJe** [comment]
@SuperSandro2000 Do you think we can finalize this PR soon?

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:26362]: I think that would be the best option we currently have.

Comment (NKJe) [reply pkgs/top-level/all-packages.nix:26362]: I've added official to the new packages, and reverted to name changes of the old packages.

**Comment by NKJe** [comment]
I just bumped the version of protonvpn-cli-official in the init commit from 3.7.1 to 3.7.2.

File: pkgs/applications/networking/protonvpn-gui-official/default.nix [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+  src = fetchFromGitHub {
+    owner = "ProtonVPN";
+    repo = "linux-app";
+    rev = version;
+    sha256 = "081f7bdbi1j3nkm0kf3flpaj0xvfwgnn0svbqpqgnqc1bjycag5y";
+  };
+
+  nativeBuildInputs = [ wrapGAppsHook gobject-introspection imagemagick ];
+
+  buildInputs = [ gtk3 ] ++ lib.optional withIndicator [ libappindicator-gtk3 ];
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/protonvpn-gui-official/default.nix:25]:
```suggestion
, libappindicator-gtk3
}:

python3Packages.buildPythonApplication rec {
  pname = "protonvpn-linux-gui-official";
  version = "1.0.1";

  src = fetchFromGitHub {
    owner = "ProtonVPN";
    repo = "linux-app";
    rev = version;
    sha256 = "081f7bdbi1j3nkm0kf3flpaj0xvfwgnn0svbqpqgnqc1bjycag5y";
  };

  nativeBuildInputs = [ wrapGAppsHook gobject-introspection imagemagick ];

  buildInputs = [ gtk3 ]
    ++ lib.optional withIndicator [ libappindicator-gtk3 ];
```

File: pkgs/applications/networking/protonvpn-gui-official/default.nix [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+    owner = "ProtonVPN";
+    repo = "linux-app";
+    rev = version;
+    sha256 = "081f7bdbi1j3nkm0kf3flpaj0xvfwgnn0svbqpqgnqc1bjycag5y";
+  };
+
+  nativeBuildInputs = [ wrapGAppsHook gobject-introspection imagemagick ];
+
+  buildInputs = [ gtk3 ]
+    ++ lib.optional withIndicator [ libappindicator-gtk3 ];
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/protonvpn-gui-official/default.nix:26]:
```suggestion
    ++ lib.optionals withIndicator [ libappindicator-gtk3 ];
```

**Comment by kiyui** [comment]
Just bumping this, I updated with the [version 1.2.0](https://github.com/ProtonVPN/linux-app/releases/tag/1.2.0) locally and it's working great. Thanks for working on this!

**Comment by NKJe** [comment]
> Just bumping this, I updated with the [version 1.2.0](https://github.com/ProtonVPN/linux-app/releases/tag/1.2.0) locally and it's working great. Thanks for working on this!

Very good. I'll bump the version once I get to civilization.

File: pkgs/development/python-modules/protonvpn-nm-lib/default.nix [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+      src = ./gi-path-patch.patch;
+      networkmanager_path = "${networkmanager}/lib/girepository-1.0";
+    })
+  ];
+
+  # Fails because it attempts to connect to DBUS
+  doCheck = false;
+
+  # Disabled because importing creates a log folder
+  # pythonImportsCheck = [ "protonvpn_nm_lib.api" ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/protonvpn-nm-lib/default.nix:50]: In $HOME? or where? Could we do something like ``export HOME=TMPDIR``?

**Comment by WolfangAukang** [comment]
Closing because of #140367, #170667, #170726.

**Pull request closed without merging**